src/xbot/common/logic/Latch.java

```diff
@@ -22,33 +22,32 @@ public Latch(boolean initialValue, EdgeType latchType) {
     public void setValue(boolean value) {
         boolean different = this.value != value;
 
-        switch (latchType) {
-        case Both:
-            if (different) {
-                saveValueAndAlertWatchers(value);
+        if (different) {
+            this.value = value;
+            switch (latchType) {
+                case Both:
+                    alertWatchers(value);
+                    break;
+                case RisingEdge:
+                    if (value) {
+                        alertWatchers(true);
+                    }
+                    break;
+                case FallingEdge:
+                    if (!value) {
+                        alertWatchers(false);
+                    }
+                    break;
+                default:
+                    break;
             }
-            break;
-        case RisingEdge:
-            if (different && value) {
-                saveValueAndAlertWatchers(value);
-            }
-            break;
-        case FallingEdge:
-            if (different && !value) {
-                saveValueAndAlertWatchers(value);
-            }
-            break;
-        default:
-            break;
         }
     }
 
-    private void saveValueAndAlertWatchers(boolean value) {
-        this.value = value;
-
+    private void alertWatchers(boolean value) {
```

Comment:
Accept edge type as parameter instead of boolean

Reply:
I just removed the alertWatchers method and handle everything in setValue